ale/drivers/cube_driver.py

```diff
@@ -0,0 +1,224 @@
+from glob import glob
+import os
+import struct
+import re
+
+import pvl
+import spiceypy as spice
+import numpy as np
+import quaternion
+
+from ale import config
+from ale.drivers.base import Base
+
+def read_table_data(table_label, file):
+    file.seek(label['StartByte']-1)
+    return file.read(label['Bytes'])
+
+def field_size(field_label):
+    data_sizes = {
+        'Integer' : 4,
+        'Double'  : 8,
+        'Real'    : 4,
+        'Text'    : 1
+    }
+    return data_sizes[field_label['Type']] * field_label['Size']
+
+def field_format(field_label):
+    data_formats = {
+        'Integer' : 'i',
+        'Double'  : 'd',
+        'Real'    : 'f'
+    }
+    return data_formats[field_label['Type']] * field_label['Size']
+
+def parse_field(field_label, data):
+    if field_label['Type'] == 'Text':
+        results[field_label['Name']].append(data[:field_label['Size']].decode(encoding='latin_1'))
+    else:
+        data_format = field_format(field_label)
+        field_data = struct.unpack_from(data_format, data)
+        if len(field_data) == 1:
+            field_data = field_data[0]
+    return field_data
+
+def parse_table_data(table_label, data):
+    fields = table_label.getlist('Field')
+    results = {field['Name']:[] for field in fields}
+    offset = 0
+    for record in range(table_label['Records']):
+        for field in fields:
+            field_data = parse_field(field, data[offset:])
+            results[field['Name']].append(field_data)
+            offset += field_size(field)
+    return results
+
+def parse_rotation_table(label, field_data):
```

Comment:
Probably worth putting some docstrings for what each if statement is actually handling. Can't tell from looking what the difference in the different J2000 strings are.